R/human_infection.R

```diff
@@ -519,13 +519,33 @@ relapse_bite_infection_hazard_resolution <- function(
   ## all bitten humans with an infectious bite (incorporating prophylaxis) get a new batch of hypnozoites
   if(bite_infections$size()>0){
     
+    ## drug prophylaxis may limit formation of new hypnozoite batches
+    ls_prophylaxis <- rep(0, bite_infections$size())
+    if(length(parameters$drug_hypnozoite_efficacy) > 0){
+      
+      ls_drug <- variables$ls_drug$get_values(bite_infections)
+      ls_medicated <- (ls_drug > 0)
+      ls_medicated[ls_drug > 0] <- !is.na(parameters$drug_hypnozoite_efficacy[ls_drug])
```

Comment:
I would expect this to fail? Since `length(ls_medicated[ls_drug > 0])` is less than or equal to `length(ls_drug)`. If some people are not medicated then there'll be a weird assignment here. Is this tested for?

```suggestion
      ls_medicated <- ls_medicated & !is.na(parameters$drug_hypnozoite_efficacy[ls_drug])
```

Reply:
I see what you're saying. An issue is that some of ls_drug will be 0 (if no drugs have been assigned to the individual), which means that `!is.na(parameters$drug_hypnozoite_efficacy[ls_drug])` may not have the same length as `ls_medicated` (as far as I can work it out). I've taken a look to make sure it's doing what it's supposed to do, with a few tests to check it through.